.github/workflows/tests_eessi_module.yml

```diff
@@ -0,0 +1,75 @@
+# documentation: https://help.github.com/en/articles/workflow-syntax-for-github-actions
+name: Tests for eessi_module_functionality in software.eessi.io
+on:
+  push:
+    branches: [ "*-software.eessi.io" ]
+  pull_request:
+permissions:
+  contents: read # to fetch code (actions/checkout)
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        EESSI_VERSION:
+          - 2023.06        
+    steps:
+      - name: Check out software-layer repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+          
+      - name: Mount EESSI CernVM-FS pilot repository
+        uses: cvmfs-contrib/github-action-cvmfs@55899ca74cf78ab874bdf47f5a804e47c198743c # v4.0
+        with:
+          cvmfs_config_package: https://github.com/EESSI/filesystem-layer/releases/download/latest/cvmfs-config-eessi_latest_all.deb
+          cvmfs_http_proxy: DIRECT
+          cvmfs_repositories: software.eessi.io
+              
+      - name: Test for archdetect_cpu functionality with only one valid path
+        run: |
+          export MODULEPATH=/cvmfs/software.eessi.io/versions/${{matrix.EESSI_VERSION}}/init/modules/:$MODULEPATH
+          CPU_ARCH=$(./init/eessi_archdetect.sh cpupath)
+          export EESSI_ARCHDETECT_OPTIONS=dummy/cpu:$CPU_ARCH:dummy1/cpu1
+          if module load EESSI/${{matrix.EESSI_VERSION}}; then
+              if [[ $CPU_ARCH == "${EESSI_SOFTWARE_SUBDIR}" ]]; then
```

Comment:
Is it intended that this would fail on an architecture like `zen4` where `eessi_archdetect.sh cpupath` would return `x86_64/amd/zen4` while `EESSI_SOFTWARE_SUBDIR` would be set to `x86_64/amd/zen3` (current redirection implemented in `archdetect_cpu` function in the lua file in this PR). For other architectures that are not present in EESSI there might be a difference too (think of `zen5`, `sapphire rapids`, `neoverse_v2`, `a64fx`, ...).

It makes me wonder a bit what this test actually tests? It looks like it tests if a software stack for the _architecture_ that `eessi_archdetect.sh cpupath` returns is present on CVMFS.

Minor issue: use (double) quotes and curly brackets around all variables.

Reply:
Merged this test along with the value comparison test when sourcing the bash script.